coalib/results/AbsolutePosition.py

```diff
@@ -0,0 +1,55 @@
+from coalib.results.TextPosition import TextPosition
+from coalib.misc.Decorators import enforce_signature
+
+
+class AbsolutePosition(TextPosition):
+
+    @enforce_signature
+    def __init__(self,
+                 text: (tuple, list, None)=None,
+                 position: (int, None)=None):
+        """
+        Creates an AbsolutePosition object that represents the index of a
+        character in a string.
+
+        :param text:     The text containing the character.
+        :param position: Position identifying the index of character
+                         in text.
+        """
+        line = column = None
+        if position and text:
+            line, column = calc_line_col(text, position)
+        self._text = text
+        self._position = position
+        super().__init__(line, column)
+
+    @property
+    def position(self):
+        return self._position
+
+
+def calc_line_col(text, position):
```

Comment:
well if we move the function into `__init__` we compromise the readability, and also the option of using it again somewhere else, i think this function can be used in a lot of places, for eg: creating a `TextPosition` object from position as you already suggested, creating `SourceRange` Objects from position and so on.